## Question 1


In [ ]:
# Programe for the numbers which are divisable by the 7 and not divisable by the 5 in between 2000 and 3200

# I am using list comprehension
# I am using logical operator AND

numbers=[i for i in range(2000,3201) if i%7==0 and i%5!=0]

In [40]:
numbers

[2002,
 2009,
 2016,
 2023,
 2037,
 2044,
 2051,
 2058,
 2072,
 2079,
 2086,
 2093,
 2107,
 2114,
 2121,
 2128,
 2142,
 2149,
 2156,
 2163,
 2177,
 2184,
 2191,
 2198,
 2212,
 2219,
 2226,
 2233,
 2247,
 2254,
 2261,
 2268,
 2282,
 2289,
 2296,
 2303,
 2317,
 2324,
 2331,
 2338,
 2352,
 2359,
 2366,
 2373,
 2387,
 2394,
 2401,
 2408,
 2422,
 2429,
 2436,
 2443,
 2457,
 2464,
 2471,
 2478,
 2492,
 2499,
 2506,
 2513,
 2527,
 2534,
 2541,
 2548,
 2562,
 2569,
 2576,
 2583,
 2597,
 2604,
 2611,
 2618,
 2632,
 2639,
 2646,
 2653,
 2667,
 2674,
 2681,
 2688,
 2702,
 2709,
 2716,
 2723,
 2737,
 2744,
 2751,
 2758,
 2772,
 2779,
 2786,
 2793,
 2807,
 2814,
 2821,
 2828,
 2842,
 2849,
 2856,
 2863,
 2877,
 2884,
 2891,
 2898,
 2912,
 2919,
 2926,
 2933,
 2947,
 2954,
 2961,
 2968,
 2982,
 2989,
 2996,
 3003,
 3017,
 3024,
 3031,
 3038,
 3052,
 3059,
 3066,
 3073,
 3087,
 3094,
 3101,
 3108,
 3122,
 3129,
 3136,
 3143,
 3157,
 3164,
 3171,
 3178,
 3192,
 3199]

In [41]:
len(numbers)

138

Conclusion : There are total 138 numbers present in between 2000 and 3200 which are divisable by 7 but not multiple of 5

## Question 2

In [ ]:
# Taking input from user 

first=input("Enter first name:")
last=input("Enter last name : ")


In [79]:
print('{} {}'.format(first,last))

jani basha


In [88]:
# concatenating first and last with slicing using
# I also concat space in between first and last

last[::-1]+ ' '+first[::-1]

'ahsab inaj'

## Question 3

In [96]:
# given diameter of the sphere is 12 cm 
# we need to find volume of the sphere 
# sphere volume formula is v=4/3*pi*r^3
# we have diameter but in formula we have radius 
# d=2r ==> r=d/2=12/2=6cm
# 4/3 constant and pi also constant 

import math  # importing math lib for pi 

diameter=input(" Enter given diamter:")

radius = int(diameter)/2

volume= 4/3*math.pi*radius**3

 Enter given diamter:16


In [97]:
volume


2144.660584850632